src/app/store/reference-data/reference-data.selectors.ts

```diff
@@ -68,19 +73,25 @@ export const selectRefDataBySearchTerm = (
 	filter: string
 ) =>
 	createSelector(referenceDataFeatureState, (state) => {
-		// eslint-disable-next-line @typescript-eslint/no-explicit-any
-		const searchItem: Array<any> = [];
+		const searchItem: ReferenceDataModelBase[] = [];
 
-		state[`${referenceDataType}`].ids.forEach((key) => {
-			// eslint-disable-next-line @typescript-eslint/no-explicit-any
-			const obj = state[`${referenceDataType}`].entities[`${key}`] as any;
-			if (obj[`${filter}`].toString().toUpperCase().includes(searchTerm.toString().toUpperCase())) {
+		state[referenceDataType]?.ids.forEach((key) => {
+			const obj = state[referenceDataType].entities[key];
+			if (
+				obj &&
+				obj[filter as keyof ReferenceDataModelBase]
+					?.toString()
+					.toUpperCase()
+					.includes(searchTerm.toString().toUpperCase())
+			) {
 				searchItem.push(obj);
 			}
 		});
+
 		if (searchTerm.length > 0) {
 			return searchItem;
 		}
+
 		return undefined;
 	});
```

Comment:
I wonder if we look at using array methods instead of pushing into `searchItem` and then checking the length. At the same time, we might then look at introducing early returns, something like 
```
export const selectRefDataBySearchTerm = (
	searchTerm: string,
	referenceDataType: ReferenceDataResourceType,
	filter: keyof ReferenceDataModelBase
) =>
	createSelector(referenceDataFeatureState, (state): ReferenceDataModelBase[] => {
		const entities = state[referenceDataType]?.entities;

		if (!searchTerm || !entities) return [];

		return Object.values(entities).filter(
			(obj): obj is ReferenceDataModelBase =>
				obj !== undefined && (obj[filter]?.toString().toUpperCase() || '').includes(searchTerm.toUpperCase())
		);
	});
```
Maybe?

Reply:
Yeah that syntax looks nicer and removes the possibility of it returning undefined, so will apply that and the stricter typings for the areas that call it.